utils/test_utils.py

```diff
@@ -0,0 +1,259 @@
+            'abcd7.gcode',
+            'abcd8.gcode',
+            'abcd9.gcode',
+            'abcd10.gcode',
+            'abcd11.gcode']
+    )
+
+def test_generate_new_filename_abc_10():
+    simulate_printing_12_times_and_validate_result(
+        filename='abc.gcode',
+        max_filename=10,
+        existing_sd_files=[],
+        expected_sd_files=[
+            'abc.gcode',
+            'abc1.gcode',
+            'abc2.gcode',
+            'abc3.gcode',
+            'abc4.gcode',
+            'abc5.gcode',
+            'abc6.gcode',
+            'abc7.gcode',
+            'abc8.gcode',
+            'abc9.gcode',
+            'ab10.gcode',
+            'ab11.gcode']
+    )
+
+def test_generate_new_filename_abc_9():
+    simulate_printing_12_times_and_validate_result(
+        filename='abc.gcode',
+        max_filename=9,
+        existing_sd_files=[],
+        expected_sd_files=[
+            'abc.gcode',
+            'ab1.gcode',
+            'ab2.gcode',
+            'ab3.gcode',
+            'ab4.gcode',
+            'ab5.gcode',
+            'ab6.gcode',
+            'ab7.gcode',
+            'ab8.gcode',
+            'ab9.gcode',
+            'a10.gcode',
+            'a11.gcode']
+    )
+
+def test_generate_new_filename_abc_8():
+    simulate_printing_12_times_and_validate_result(
+        filename='abc.gcode',
+        max_filename=8,
+        existing_sd_files=[],
+        expected_sd_files=[
+            'ab.gcode',
+            'a1.gcode',
+            'a2.gcode',
+            'a3.gcode',
+            'a4.gcode',
+            'a5.gcode',
+            'a6.gcode',
+            'a7.gcode',
+            'a8.gcode',
+            'a9.gcode',
+            '10.gcode',
+            '11.gcode']
+    )
+
+def test_generate_new_filename_abc_7():
+    simulate_printing_12_times_and_validate_result(
+        filename='abc.gcode',
+        max_filename=7,
+        existing_sd_files=[],
+        expected_sd_files=[
+            'a.gcode',
+            '1.gcode',
+            '2.gcode',
+            '3.gcode',
+            '4.gcode',
+            '5.gcode',
+            '6.gcode',
+            '7.gcode',
+            '8.gcode',
+            '9.gcode',
+            '10.gcode',
+            '11.gcode']
+    )
+
+def test_generate_new_filename_abc_6():
+    simulate_printing_12_times_and_validate_result(
+        filename='abc.gcode',
+        max_filename=6,
+        existing_sd_files=[],
+        expected_sd_files=[
+            '.gcode',
+            '1.gcode',
+            '2.gcode',
+            '3.gcode',
+            '4.gcode',
+            '5.gcode',
+            '6.gcode',
+            '7.gcode',
+            '8.gcode',
+            '9.gcode',
+            '10.gcode',
+            '11.gcode']
+    )
+
+def test_generate_new_filename_abc_5():
+    simulate_printing_12_times_and_validate_result(
+        filename='abc.gcode',
+        max_filename=5,
+        existing_sd_files=[],
+        expected_sd_files=[
+            '.gcode',
```

Comment:
it looks like not a good behavior, we can't use filename only with the filename extension

Reply:
yeah, it might look not so good, but it is a valid filename :)
![image](https://user-images.githubusercontent.com/5782954/149216807-daf33f2b-b077-4914-93f3-87dcc4b7c544.png)

Reply:
ok, then. It's a trifle anyway, let's keep it as it is. If there will some sort of negotiation from the community it will be easier to place at the separate task